### ISSUE TITLE
Doubts about engine future

### ISSUE THREAD

**Issue by aureliosaraiva** [resoai/TileBoard#629]
Hello guys, first congratulations on the project.

I was analyzing the source code and saw that the project uses 1.8 angular. I used it for a long time, I really like it. But I wonder if it would be better to use a newer technology before having a project destination.

Are there already conversations to exchange the line for a more recent one like React, vue.js or even Angular 11?

**Comment by rchl** [comment]
There is no reason to change the technology used just because it's old. Angular 1.x works well, is still supported by Google, is compatible both with old and new browsers, and is easy to work with (compared to the "new" angular).

The only reason we could consider switching to something else is if it would give significant improvements in some areas but I can't really see any such major reason (at least compared to Vue, which I'm most familiar with).

It would also many many people annoyed as they might have customized parts of the code or CSS and any such breaking changes would be a hurdle for them. Some people just want stuff to work and they don't have time to keep fixing their smart home. :)

**Comment by rchl** [comment]
Apparently, the support for AngularJS ends on December 31, 2021 ([source](https://docs.angularjs.org/misc/version-support-status)) so then it might be a good time to start to think about switching to something else. :)

**Comment by aureliosaraiva** [comment]
> There is no reason to change the technology used just because it's old. Angular 1.x works well, is still supported by Google, is compatible both with old and new browsers, and is easy to work with (compared to the "new" angular).
> 
> The only reason we could consider switching to something else is if it would give significant improvements in some areas but I can't really see any such major reason (at least compared to Vue, which I'm most familiar with).
> 
> It would also annoy many people as they might have customized parts of the code or CSS and any such breaking changes would be a hurdle for them. Some people just want stuff to work and they don't have time to keep fixing their smart home. :)

I agree with your first comment. It was much more a curiosity. The exchange could be quite painful for many people.

I have been working with JavaScript for many years now, and for some time now I have seen a massive migration to more modern technologies.





> Apparently, the support for AngularJS ends on December 31, 2021 ([source](https://docs.angularjs.org/misc/version-support-status)) so then it might be a good time to start to think about switching to something else. :)

About the second comment :joy::joy::joy:,. It was what motivated me to question whether there was any migration plan, since it will at some point be discontinued.

**Comment by rchl** [comment]
> It was what motivated me to question whether there was any migration plan, since it will at some point be discontinued.

We don't have any plan and also don't have that much time to work on it. :)
So we'll see how it goes. Likely (hopefully?) AngularJS won't break shortly after not being supported. :)

**Comment by alphasixtyfive** [comment]
Amazon's Fire TV & Tablets are still based on Android 5 if I'm not mistaken. If you have some free time you are more than welcome to contribute. I think that there are a lot of things with higher priority which could be implemented before abandoning rusty old Angular 1.x.

**Comment by aureliosaraiva** [comment]
> Amazon's Fire TV & Tablets are still based on Android 5 if I'm not mistaken. If you have some free time you are more than welcome to contribute. I think that there are a lot of things with higher priority which could be implemented before abandoning rusty old Angular 1.x.

in fact, I even have time to start looking for what exists today.  what would be the process?

**Comment by alphasixtyfive** [comment]
> > Amazon's Fire TV & Tablets are still based on Android 5 if I'm not mistaken. If you have some free time you are more than welcome to contribute. I think that there are a lot of things with higher priority which could be implemented before abandoning rusty old Angular 1.x.
> 
> in fact, I even have time to start looking for what exists today. what would be the process?

We always wanted to scrap current light tile and do one based on the newly made popup. There a re few of minor-fish things in the issues, like making `TEXT_LIST` tile accept dynamic `list` and `DEVICE_TRAKER` tile with map popup. There is also a lot of refactoring and code splitting and cleanup to be done in general as the project was never designed to be what it is today, it started as a single webpage with three different tiles for my hallway tablet :)
